src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -38,17 +41,31 @@ public class DeletionTime implements Comparable<DeletionTime>, IMeasurableMemory
     /**
      * A special DeletionTime that signifies that there is no top-level (row) tombstone.
      */
-    public static final DeletionTime LIVE = new DeletionTime(Long.MIN_VALUE, Integer.MAX_VALUE);
+    public static final DeletionTime LIVE = new DeletionTime(Long.MIN_VALUE, Long.MAX_VALUE);
 
     public static final Serializer serializer = new Serializer();
+    public static final Serializer legacySerializer = new LegacySerializer();
 
     private final long markedForDeleteAt;
-    private final int localDeletionTime;
+    protected final int localDeletionTimeGuavaUint;
+
+    public static DeletionTime getInstance(long markedForDeleteAt, long localDeletionTime)
```

Comment:
We frequently use the name `getInstance` to either access singletons or retrieve cached instances (`MapType`, `ListType`, `CompositeType`, `ToJsonFct`, etc.). However this factory method always returns a new instance, so maybe we could name it `build` instead.